src/core/diags/factory_diags.cpp

```diff
@@ -274,7 +275,8 @@ Error Diags::ProcessSend(uint8_t aArgsLength, char *aArgs[], char *aOutput, size
     SuccessOrExit(error = ParseLong(aArgs[1], value));
     VerifyOrExit(value <= OT_RADIO_FRAME_MAX_SIZE, error = kErrorInvalidArgs);
     VerifyOrExit(value >= OT_RADIO_FRAME_MIN_SIZE, error = kErrorInvalidArgs);
-    mTxLen = static_cast<uint8_t>(value);
+    mTxLen      = static_cast<uint8_t>(value);
+    mDiagSendOn = true;
```

Comment:
The function `ProcessSend()` could also send the radio packets. It is better to set the flag `mDiagSendOn` before calling the function `Get<Radio>().Transmit()` and clear the  flag `mDiagSendOn` in the function `Diags::TransmitDone()`.